### PR TITLE
Add Vary header for Hx-Requests

### PR DESCRIPTION
**Pull request by dsyer**
The HTMX docs recommend doing this to avoid issues with caches returning partial results. https://htmx.org/docs/#caching

### CODE REVIEW

**Comment by wimdeblauwe** [comment]
Thanks! I see the build is failing, because `/hx-vary` endpoint does not exist. Did you forget to commit something?

File: htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxHandlerInterceptor.java [hunk 1]
```diff
@@ -23,11 +24,16 @@ public boolean preHandle(HttpServletRequest request,
...
             setHxRefresh(response, method);
+            setVary(response, method);
         }
 
         return true;
     }
 
+    private void setVary(HttpServletResponse response, Method method) {
+        response.addHeader(HttpHeaders.VARY, HtmxRequestHeader.HX_REQUEST.getValue());
+    }
```
Comment (xhaggi) [inline htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxHandlerInterceptor.java:35]: Correct me if I'm wrong, but this adds `HX-Request` to the `VARY` header for all responses, whether it's necessary or not. Is that what we want?

**Comment by wimdeblauwe** [comment]
Thanks for your remark @xhaggi ! @dsyer Is it ok for you to merge now?

**Comment by dsyer** [comment]
Yes please.

**Pull request merged**